### DISCUSSION TITLE
[feature] Edit config from tray context menu

### DISCUSSION THREAD

**Discussion by karanrajpal14** [espanso/espanso#1195] [Feature requests and Ideas]
Hello,

Currently, we do have options to Disable, Reload Config and Exit in the system tray menu. Is there any chance you could add another menu item to the tray menu for editing the config which opens up the `default.yml` in the default text editor?

**Comment by federico-terzi** [comment]
Hey @karanrajpal14,

I agree with you, this feature would be nice to have. I was planning to add it when the GUI editor is ready, so stay tuned :)

**Comment by oaustegard** [comment]
Somewhat related: is there a way to make espanso honor the system default for .yml file extensions in Windows? ~Currently `espanso edit` opens in Notepad no matter what I do it seems - is the expectation that a user has to set a System Environment Variable named "EDITOR" with the path to the preferred editor?~ Indeed it is - and it is documented: https://espanso.org/docs/configuration/#customizing-the-editor
https://github.com/federico-terzi/espanso/blob/279aace2bbaaaa4d273d53c11baa92ccc2306854/src/edit.rs#L39

Fixed this with a `set "EDITOR=C:\Users\austegard.oskar\AppData\Local\Programs\Microsoft VS Code\Code.exe"` from the command prompt.

Still might be a good idea to instead implement https://docs.rs/opener/0.4.1/opener/ or similar to allow yml files to be edited with the expected editor?

**Comment by federico-terzi** [comment]
@oaustegard Glad you figured that out!

> Still might be a good idea to instead implement https://docs.rs/opener/0.4.1/opener/ or similar to allow yml files to be edited with the expected editor?

Keep in mind that espanso was designed to be a terminal-first app in the beginning, so on Linux and macOS `espanso edit` either opens `nano` or `vim`. On windows, because there is no default terminal text editor, I defaulted to Notepad.

In the future, espanso will ship with a GUI that will make this process (hopefully) much easier :)

**Comment by firinael** [comment]
> Somewhat related: is there a way to make espanso honor the system default for .yml file extensions in Windows? Currently espanso edit opens in Notepad no matter what I do it seems - is the expectation that a user has to set a System Environment Variable named "EDITOR" with the path to the preferred editor?

I've had issues with this twice already and had to go on a quest to find out that you need to set an `EDITOR` environment variable. Please make this very clear in the docs and add some user-friendly way to specify your chosen editor.

**Comment by oaustegard** [comment]
I have since taken this a step further, creating a trigger/match that opens a Vs Code workspace that contains the yams as well as the various scripts I use with espanso. Not for the masses probably, but great for when needing to make tweaks.
